libtenzir/builtins/aggregation-functions/all.cpp

```diff
@@ -51,7 +56,43 @@ class all_function final : public aggregation_function {
   std::optional<bool> all_ = {};
 };
 
-class plugin : public virtual aggregation_function_plugin {
+class all_instance final : public aggregation_instance {
+public:
+  explicit all_instance(ast::expression expr) : expr_{std::move(expr)} {
+  }
+
+  auto update(const table_slice& input, session ctx) -> void override {
+    if (failed) {
+      return;
+    }
+    auto arg = eval(expr_, input, ctx);
+    auto f = detail::overload{
+      [](const arrow::NullArray&) {},
```

Comment:
This needs the same behavior as a `null` of `bool_type`.

Reply:
Should just receiving a `NullArray` be failure or it must have at least one element?

Reply:
Hmm, good question. I think you can just assume that it's non-empty.